scijava-function/src/main/java/org/scijava/function/Computers.java

```diff
@@ -419,4 +552,2339 @@ default void accept(final I1 in1, final I2 in2, final I3 in3, final I4 in4,
 		}
 	}
+
+	@FunctionalInterface
+	public interface Arity1_1<O, I> extends Arity1<I, O> {
+
+		void compute1(@Container O out, I in);
+
+		@Override
+		default void compute(I in, @Container O out) {
+			compute1(out, in);
+		}
+	}
+
+	@FunctionalInterface
+	public interface Arity2_1<O, I2, I1> extends Arity2<I1, I2, O> {
+
+		void compute1(@Container O out, I2 in2, I1 in1);
+
+		@Override
+		default void compute(I1 in1, I2 in2, @Container O out) {
+			compute1(out, in2, in1);
+		}
+	}
+
+	@FunctionalInterface
+	public interface Arity2_2<I1, O, I2> extends Arity2<I1, I2, O> {
+
+		void compute2(I1 in1, @Container O out, I2 in2);
+
+		@Override
+		default void compute(I1 in1, I2 in2, @Container O out) {
+			compute2(in1, out, in2);
+		}
+	}
+
+	@FunctionalInterface
+	public interface Arity3_1<O, I2, I3, I1> extends Arity3<I1, I2, I3, O> {
+
+		void compute1(@Container O out, I2 in2, I3 in3, I1 in1);
+
+		@Override
+		default void compute(I1 in1, I2 in2, I3 in3, @Container O out) {
+			compute1(out, in2, in3, in1);
+		}
+	}
+
+	@FunctionalInterface
+	public interface Arity3_2<I1, O, I3, I2> extends Arity3<I1, I2, I3, O> {
+
+		void compute2(I1 in1, @Container O out, I3 in3, I2 in2);
+
+		@Override
+		default void compute(I1 in1, I2 in2, I3 in3, @Container O out) {
+			compute2(in1, out, in3, in2);
+		}
+	}
+
+	@FunctionalInterface
+	public interface Arity3_3<I1, I2, O, I3> extends Arity3<I1, I2, I3, O> {
+
+		void compute3(I1 in1, I2 in2, @Container O out, I3 in3);
+
+		@Override
+		default void compute(I1 in1, I2 in2, I3 in3, @Container O out) {
+			compute3(in1, in2, out, in3);
+		}
+	}
+
+	@FunctionalInterface
+	public interface Arity4_1<O, I2, I3, I4, I1> extends
+		Arity4<I1, I2, I3, I4, O>
+	{
+
+		void compute1(@Container O out, I2 in2, I3 in3, I4 in4, I1 in1);
+
+		@Override
+		default void compute(I1 in1, I2 in2, I3 in3, I4 in4, @Container O out) {
+			compute1(out, in2, in3, in4, in1);
+		}
+	}
+
+	@FunctionalInterface
+	public interface Arity4_2<I1, O, I3, I4, I2> extends
+		Arity4<I1, I2, I3, I4, O>
+	{
+
+		void compute2(I1 in1, @Container O out, I3 in3, I4 in4, I2 in2);
+
+		@Override
+		default void compute(I1 in1, I2 in2, I3 in3, I4 in4, @Container O out) {
+			compute2(in1, out, in3, in4, in2);
+		}
+	}
+
+	@FunctionalInterface
+	public interface Arity4_3<I1, I2, O, I4, I3> extends
+		Arity4<I1, I2, I3, I4, O>
+	{
+
+		void compute3(I1 in1, I2 in2, @Container O out, I4 in4, I3 in3);
+
+		@Override
+		default void compute(I1 in1, I2 in2, I3 in3, I4 in4, @Container O out) {
+			compute3(in1, in2, out, in4, in3);
+		}
+	}
+
+	@FunctionalInterface
+	public interface Arity4_4<I1, I2, I3, O, I4> extends
+		Arity4<I1, I2, I3, I4, O>
+	{
+
+		void compute4(I1 in1, I2 in2, I3 in3, @Container O out, I4 in4);
+
+		@Override
+		default void compute(I1 in1, I2 in2, I3 in3, I4 in4, @Container O out) {
+			compute4(in1, in2, in3, out, in4);
+		}
+	}
+
+	@FunctionalInterface
+	public interface Arity5_1<O, I2, I3, I4, I5, I1> extends
```

Comment:
This should be `<O, I1, I2, I3, I4, I5>`, shouldn't it? Right now it looks like it just swapped the displaced I param with the O, but it should shunt everything over, no?

Reply:
Fixed in e843de3